dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/SelectGuestBookService.java

```diff
@@ -0,0 +1,60 @@
+package edu.harvard.iq.dataverse.guestbook;
+
+import edu.harvard.iq.dataverse.dataset.datasetversion.DatasetVersionServiceBean;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.guestbook.Guestbook;
+import io.vavr.control.Option;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.sql.Timestamp;
+import java.time.Clock;
+import java.time.Instant;
+
+@Stateless
+public class SelectGuestBookService {
+
+    private DatasetVersionServiceBean versionService;
+    private Clock systemTime = Clock.systemDefaultZone();
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    @Deprecated
+    public SelectGuestBookService() {
+    }
+
+    @Inject
+    public SelectGuestBookService(DatasetVersionServiceBean versionService) {
+        this.versionService = versionService;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public Dataset saveGuestbookChanges(DatasetVersion editedDataset,
+                        Option<Guestbook> selectedGuestbook,
+                        Option<Guestbook> guestbookBeforeChanges) {
+
+        if (isGuestbookAddedOrRemoved(selectedGuestbook, guestbookBeforeChanges)){
+            Dataset dataset = editedDataset.getDataset();
```

Comment:
Maybe it would be better to place this variable before if, so that we can remove it from line 43.